<a href="https://colab.research.google.com/github/hachristina/Project2/blob/main/Project_2_Part_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Task
Your task for this week is to propose two possible datasets you would like to work with for Project 2.  

For each possible dataset, show the .head() of the dataset and answer the following questions:

1) Source of data

2) Brief description of data

3) What is the target?

4) Is this a classification or regression problem?

5) How many features?

6) How many rows of data.

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

#Dataset 1: Board Games

In [15]:
bg_df = pd.read_csv("/content/drive/MyDrive/Data Science Bootcamp/Project 2/bgg_dataset.csv", sep=';')
bg_df.head()

ID  ...                         Domains
0  174430.0  ...  Strategy Games, Thematic Games
1  161936.0  ...  Strategy Games, Thematic Games
2  224517.0  ...                  Strategy Games
3  167791.0  ...                  Strategy Games
4  233078.0  ...  Strategy Games, Thematic Games

[5 rows x 14 columns]

In [16]:
bg_df.shape

(20343, 14)

In [17]:
bg_df.isnull().any()

ID                     True
Name                  False
Year Published         True
Min Players           False
Max Players           False
Play Time             False
Min Age               False
Users Rated           False
Rating Average        False
BGG Rank              False
Complexity Average    False
Owned Users            True
Mechanics              True
Domains                True
dtype: bool

1) Source of data
https://www.kaggle.com/andrewmvd/board-games?select=bgg_dataset.csv

Citation
Dilini Samarasinghe, July 5, 2021, "BoardGameGeek Dataset on Board Games", IEEE Dataport, doi: https://dx.doi.org/10.21227/9g61-bs59.

2) Brief description of data
This dataset contains data collected on board games from the BoardGameGeek (BGG) website in February 2021.

3) What is the target?
Board game average rating

4) Is this a classification or regression problem?
Regression

5) How many features?
14

6) How many rows of data.
20343

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
There are null values to address in this dataset

#Dataset 2: World Happiness Report 2021

In [1]:
import pandas as pd

In [2]:
whr_df = pd.read_csv('/content/drive/MyDrive/Data Science Bootcamp/Project 2/world-happiness-report.csv')
whr_df.head()

Country name  year  ...  Positive affect  Negative affect
0  Afghanistan  2008  ...            0.518            0.258
1  Afghanistan  2009  ...            0.584            0.237
2  Afghanistan  2010  ...            0.618            0.275
3  Afghanistan  2011  ...            0.611            0.267
4  Afghanistan  2012  ...            0.710            0.268

[5 rows x 11 columns]

In [4]:
whr_df.shape

(1949, 11)

In [9]:
whr_df.isnull().any()

Country name                        False
year                                False
Life Ladder                         False
Log GDP per capita                   True
Social support                       True
Healthy life expectancy at birth     True
Freedom to make life choices         True
Generosity                           True
Perceptions of corruption            True
Positive affect                      True
Negative affect                      True
dtype: bool

1) Source of data

https://www.kaggle.com/ajaypalsinghlo/world-happiness-report-2021/version/2?select=world-happiness-report.csv

2) Brief description of data

This dataset shows happiness scores and rankings of countries around the world between 2005-2020.


3) What is the target?

The target is the positive affect score


4) Is this a classification or regression problem?

Regression


5) How many features?

11


6) How many rows of data?

1949


7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

It looks like there will be null values to address.